packages/expect/src/matchers.js

```diff
@@ -546,22 +532,36 @@ const matchers: MatchersObject = {
           `Not to have a nested property:\n` +
           `  ${printExpected(keyPath)}\n` +
           (valuePassed ? `With a value of:\n  ${printExpected(value)}\n` : '')
-      : () =>
-          matcherHint('.toHaveProperty', 'object', 'path', {
-            secondArgument: valuePassed ? 'value' : null,
-          }) +
-          '\n\n' +
-          `Expected the object:\n` +
-          `  ${printReceived(object)}\n` +
-          `To have a nested property:\n` +
-          `  ${printExpected(keyPath)}\n` +
-          (valuePassed ? `With a value of:\n  ${printExpected(value)}\n` : '') +
-          (traversedPath
-            ? `Received:\n  ${RECEIVED_COLOR(
-                'object',
-              )}.${traversedPath}: ${printReceived(lastTraversedObject)}`
-            : '') +
-          (diffString ? `\nDifference:\n\n${diffString}` : '');
+      : () => {
+          const diffString =
+            valuePassed && hasEndProp
+              ? diff(value, result.value, {
+                  expand: this.expand,
+                })
+              : '';
```

Comment:
Can we put this in one line for readability?
```js
? diff(value, result.value, {expand: this.expand})
: '';
```

Reply:
Yes, good call.